src/Client.js

```diff
@@ -1,6 +1,33 @@
 import AMQP from './network/AMQP';
 import defaultConfig from './config/default.json';
 
+const promisifyRequest = async (amqp, options) => {
+  const {
+    exchangeIn,
+    keyIn,
+    message,
+    headers,
+    exchangeOut,
+    keyOut,
+  } = options;
+  const queue = `client-${exchangeOut}-${keyOut}`;
+  const consumerTag = `consumer-${exchangeOut}-${keyOut}`;
+
+  return new Promise((resolve, reject) => {
+    try {
+      // OBS: subscribeTo is an async method
+      amqp.subscribeTo(exchangeOut, [keyOut], queue, (msg) => {
+        resolve(msg);
```

Comment:
Please, move to the bottom.

```suggestion
        return resolve(msg);
```

Reply:
This `return` is necessary? I just move the `resolve` function to the end